### DISCUSSION TITLE
Deprecate rarely used / old / dysfunctional E2 types

### DISCUSSION THREAD

**Discussion by vurvdev** [wiremod/wire#2785] [Suggestions]
`matrix`, `matrix2` `matrix4` and `vector4` should really not exist, at least in their current states.

## Matrices
I don't think a single person has used E2 matrices in years.
I couldn't find any issues or pull requests even as little as mentioning them.
Quaternions are king with Gohidas' Physics guide.

Better matrix type would just bind to a [VMatrix](https://wiki.facepunch.com/gmod/Global.Matrix).

## Vector4

Same as in https://github.com/wiremod/wire/discussions/2721

I'd add `vector2`, but it's commonly used for positions in EGP, and there's no good alternative (no fixed size array type)

**Comment by Denneisk** [comment]
A replacement for vector2 would require rewriting a good bit of everything that assumes number values. Would be more than fixed size, but also fixed type.

Could see non-indexed EGP create methods using Vectors with Z for depth, but that might be a bit awkward.

**Comment by CornerPin** [comment]
`vector4` is also used to set UV coordinates for EGP polygon vertices, perhaps it could be replaced with a new `vertex` type?

**Reply by vurvdev** [reply]
Did you mean vector2? Cause I didn't include that in this discussion for this reason

**Reply by CornerPin** [reply]
No, I meant that besides RGBA color, `vector4` is also used in `egpPoly`, `egpPolyOutline`, and `egpLineStrip` to provide `x`, `y`, `u`, and `v` coordinates of polygon/line vertices.
https://github.com/wiremod/wire/blob/56ee9b9b609d24420c0321dacd35fc531e7babee/lua/wire/client/e2descriptions.lua#L1494-L1499

**Reply by vurvdev** [reply]
From the snippet you gave, all of those functions could just be passed an array including four numbers and it wouldn't know the difference, and I feel like that's a rare enough of a use to allow a breaking change for.

That function taking a vector4 instead of an array is also a major hack
